runtimes/polimec/src/xcm_config.rs

```diff
@@ -226,7 +226,37 @@ match_types! {
 		MultiLocation { parents: 1, interior: X1(Parachain(_)) }
 	};
 }
-
+use polimec_xcm_executor::polimec_traits::OnResponse;
+/// Allows only messages if the generic `ResponseHandler` expects them via `expecting_response`.
+pub struct AllowKnownQueryResponses<ResponseHandler>(PhantomData<ResponseHandler>);
```

Comment:
Why do we need this custom struct? Is the xcm-builder type not correct?

Reply:
It's bugged. I'll submit an issue soon.
In short, they made XCM router by default to append a `SetTopic` at the end of every sent message, and so a query response will contain 2 messages. The default `AllowKnownQueryResponses` expects only 1.